> Hope you don't mind me making changes; looks like another test needed updating. :)

Not at all, good catch. I didn't look closely at all of the tests and that one was still passing :facepalm:

Before your change I get:

```py3
======================================================================
ERROR: test_decrypt_raises_value_error_short_secret2 (tests.TestSplunkSecrets)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/workspaces/splunksecrets/tests.py", line 140, in test_decrypt_raises_value_error_short_secret2
    splunksecrets.decrypt(
  File "/workspaces/splunksecrets/splunksecrets.py", line 86, in decrypt
    plaintext = decryptor.update(ciphertext).decode()
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xee in position 1: invalid continuation byte

----------------------------------------------------------------------
```

when commenting out the `with self.assertRaises(ValueError)` so it was "passing" but not correctly. I'll open a separate PR to update the tests for the pre-7.2 tests so the same doesn't happen with those.

--- a/splunksecrets.py
+++ b/splunksecrets.py
@@ -63,7 +63,7 @@ def decrypt(secret, ciphertext, nosalt=False):
         plaintext = "".join([six.unichr(c) for c in chars])
     elif ciphertext.startswith("$7$"):
         # pad secret to 254 bytes with nulls
-        secret = secret.ljust(254, b"\0")
+        secret = six.ensure_binary(secret).ljust(254, b"\0")
 
         ciphertext = b64decode(ciphertext[3:])
 
@@ -121,7 +121,7 @@ def encrypt(secret, plaintext, nosalt=False):
 def encrypt_new(secret, plaintext, iv=None):  # pylint: disable=invalid-name
     """Use the new AES 256 GCM encryption in Splunk 7.2"""
     # pad secret to 254 bytes with nulls
-    secret = secret.ljust(254, b"\0")
+    secret = six.ensure_binary(secret).ljust(254, b"\0")
 
     kdf = PBKDF2HMAC(
         algorithm=hashes.SHA256(),

--- a/splunksecrets.py
+++ b/splunksecrets.py
@@ -357,6 +357,8 @@ def __load_splunk_secret(ctx, param, value):  # pragma: no cover
     if ctx.get_parameter_source(param.name).name != "ENVIRONMENT":
         with open(value, "rb") as f:  # pylint: disable=invalid-name
             value = f.read().strip()
+    elif isinstance(value, str):
+        value = bytes(value, encoding="utf-8")
 
     return value.strip()
 

--- a/tests.py
+++ b/tests.py
@@ -141,7 +141,6 @@ class TestSplunkSecrets(unittest.TestCase):
         )
         self.assertEqual(plaintext, "short123")
 
-
     def test_decrypt_nosalt(self):
         plaintext = splunksecrets.decrypt(splunk_secret, "$1$2+1yGuQ1gcMK", nosalt=True)
         self.assertEqual(plaintext, "temp1234")

--- a/tests.py
+++ b/tests.py
@@ -105,6 +105,14 @@ class TestSplunkSecrets(unittest.TestCase):
         )
         self.assertEqual(ciphertext, "$7$aTVkS01HYVNJUk5wSnR5NKR+EdOfT4t84WSiXvPFHGHsfHtbgPIL3g==")
 
+    def test_encrypt_new_str_secret(self):
+        ciphertext = splunksecrets.encrypt_new(
+            "abc123",  # secret as a string (not bytes)
+            "strings are fine",
+            iv=six.b("zIDM0YmIgDQ2gMzk")
+        )
+        self.assertEqual(ciphertext, "$7$eklETTBZbUlnRFEyZ016a+2HMVEtbCAJkNb7RkVHdqZwZkVJyZ+HmTWlYJedFdR4")
+
     def test_encrypt_new_pads_short_secret(self):
         ciphertext = splunksecrets.encrypt_new(
             splunk_secret[:30],

